tests/test_utils.go

```diff
@@ -260,32 +267,43 @@ var (
 )
 
 // WithClock sets clock to context.
-func WithClock(ctx context.Context, t *testing.T) context.Context {
-	t.Helper()
+func WithClock(ctx context.Context, path string) (context.Context, error) {
 	var fc clockwork.FakeClock
+	var err error
 	if GetRecording() != ModeReplaying {
-		fc = SetClock(t)
+		fc, err = SetClock(path)
 	} else {
-		fc = RestoreClock(t)
+		fc, err = RestoreClock(path)
 	}
-	return context.WithValue(ctx, clockKey, fc)
+	if err != nil {
+		return nil, err
+	}
+	return context.WithValue(ctx, clockKey, fc), nil
 }
 
 // UniqueEntityName will return a unique string that can be used as a title/description/summary/...
 // of an API entity. When used in Azure Pipelines and RECORD=true or RECORD=none, it will include
 // BuildId to enable mapping resources that weren't deleted to builds.
 func UniqueEntityName(ctx context.Context, t *testing.T) *string {
+	name := WithUniqueSurrounding(ctx, t.Name())
```

Comment:
Why the wrapping instead of putting the logic of "WithUniqueSurrounding" here?

Reply:
I didn't want to change the current usage that passes `t` instead of `name`.